contributors/devel/api_changes.md

```diff
@@ -347,21 +348,25 @@ before starting "all the rest".
 
 ### Edit types.go
 
-The struct definitions for each API are in `pkg/api/<version>/types.go`. Edit
-those files to reflect the change you want to make. Note that all types and
-non-inline fields in versioned APIs must be preceded by descriptive comments -
-these are used to generate documentation. Comments for types should not contain
-the type name; API documentation is generated from these comments and end-users
-should not be exposed to golang type names.
+The struct definitions for each API are in
+`staging/src/k8s.io/api/<group>/<version>/types.go`. Edit those files to reflect
```

Comment:
Added description on `+k8s:deepcopy-gen`.

Only people who add new API group or version need to know the [tags in doc.go](https://github.com/kubernetes/kubernetes/blob/master/pkg/api/v1/doc.go#L17-L20). In the "adding new API" sections, the instruction is copy the existing group and replace the "group" and "version" strings, that should cover all such tags. I think people won't appreciate long explanations on each of those tags. Also IMHO most tags should be moved to the build system and handled implicitly.

For the tags on the types, i.e., `genclient` and `deepcopy-gen`, the doc explained them.